operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/EventProcessor.java

```diff
@@ -88,11 +89,14 @@ private EventProcessor(ResourceCache<R> resourceCache, ExecutorService executor,
     this.retry = retry;
     this.resourceCache = resourceCache;
     this.metrics = metrics != null ? metrics : Metrics.NOOP;
-    this.eventMarker = eventMarker;
+    this.eventMarker = new EventMarker();
+    this.retryAndRescheduleTimerEventSource = new TimerEventSource<>();
```

Comment:
I don't know, but the thinking was that EventSourceManager encapsulates all the event sources, so manages also the lifecycle etc of the. This breaks that logic.

Reply:
I understand but that makes the code more complex. I tend to think that things that are only needed in one spot should be managed there.

Reply:
This is kinda a `double-edged sword`, since on the other hand it brings complexity in terms of managing the life cycle of an event source in the processor, which is kinda smell if we have already a manager for them. So those are managed in one place.

Reply:
The problem is that `EventManager` and `EventProcessor` should really be a single object as is evidenced by their tied lifecycle and the need to set a reciprocal relations between them. If we insist on splitting them, then they should each manage the parts that only they use.

Reply:
Reverted.